model/writer.go

```diff
@@ -628,13 +628,16 @@ func procPage(p *PdfPage) {
 	}
 
 	// Add font as needed.
-	f := DefaultFont()
-	p.Resources.SetFontByName("UF1", f.ToPdfObject())
+	fontName := core.PdfObjectName("UF1")
+	if d, ok := core.GetDict(p.Resources.Font); ok && d.Get(fontName) == nil {
```

Comment:
If it already exists would it make sense to check that it is the same font as intended and if not, use a different name ?  E.g. UF2 ?  Presumably this case is assuming that the file has been processed multiple times by unidoc and the font has been created multiple times?

Reply:
No, the issue is when processing files that haven't been created using unipdf. The Font dictionary of the page Resources is not recognized as being modified by the writer when adding UF1. Because of this, the reference to the UF1 font is never updated. It remains 0. 

Not sure it's worth verifying that the font is the same as the one that's added by the procPage method. Maybe we should just use a font name that's very unlikely to be used.

Reply:
OK, I think UF1 is probably already pretty uncommon, so this is fine.